### PR TITLE
expose adData during ad_start event

### PR DESCRIPTION
**Pull request by Cycymomo**
During `ad_start` event, the `adData` Object is available in the `player` instance

- create a Player
```
<div id="player"></div>
<script src="https://api.dmcdn.net/all.js"></script>

const player = DM.player('player', { video: 'x730nnd' })
```

- attach the listener to it
```
player.addEventListener('ad_start', () => {
  console.log(player.adData)
})
```
=> as soon as the ad is displayed, you'll have the `adData` informations in the `player` instance

### CODE REVIEW

File: src/core/player.js [hunk 1]
```diff
@@ -363,14 +363,14 @@ DM.provide('Player',
...
             case 'controlschange': this.controls = DM.parseBool(event.controls); break;
             case 'volumechange': this.volume = parseFloat(event.volume); this.muted = DM.parseBool(event.muted); break;
+            case 'ad_start': this.adData = event.adData;
             case 'video_start':
-            case 'ad_start':
             case 'ad_play':
             case 'playing':
             case 'play': this.paused = false; break;
             case 'end': this.ended = true; break; // no break, also set paused
+            case 'ad_end': this.adData = null;
```
Comment (dmail) [inline src/core/player.js:372]: Should we do the same on `error` event ? Have you verified that `ad_end` is dispatched when `ad_error` occurs ? (I am pretty sure yes but we need to check).

Comment (dharFr) [reply src/core/player.js:372]: Good point. 
I'd say it should be consistent with NPE implementation (e.g. when do we clean it in the player state?)

Comment (Cycymomo) [reply src/core/player.js:372]: after verification (thx Dam), `ad_end` is well dispatched after an error

**Review by dmail — APPROVED** [review]

**Comment by dmail** [comment]
Pull request was missing a gif and a link to the jira ticket by the way :)

**Pull request merged**